clio-server/src/test/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAOSpec.scala

```diff
@@ -61,4 +60,39 @@ class PersistenceDAOSpec extends AsyncFlatSpec with Matchers {
       succeed
     }
   }
+
+  it should "return metadata documents from GCS in order" in {
+    val dao = new MemoryPersistenceDAO()
+    val half = 5
+    val documents = (0L until (2L * half)).map(
+      n =>
+        DocumentMock.default.copy(
+          mockKeyLong = n,
+          mockFilePath = Some(s"gs://document-mock-key-${n}")
+      )
+    )
+    val expected = documents.drop(half)
+    val result = dao
+      .initialize(index)
+      .map(
+        _ =>
+          documents
+            .map(document => dao.writeUpdate(document, index))
+      )
+      .flatMap(_ => dao.getAllSince(expected.head.upsertId, index))
+    documents.toSet.size should be(documents.size)
+    documents.map(_.upsertId).toSet.size should be(documents.size)
```

Comment:
It looks like these two asserts could be done on line 74 above, just after `documents` has been created, as they don't involve `expected` or `result`.

Reply:
OK. I liked all the asserts at the bottom where I expect them, but YMMV.
They are kind of redundant anyway, but I moved them earlier.